src/lib/operators/index_scan.cpp

```diff
@@ -42,12 +42,16 @@ std::shared_ptr<const Table> IndexScan::_on_execute() {
   if (_included_chunk_ids.empty()) {
     jobs.reserve(_in_table->chunk_count());
     for (auto chunk_id = ChunkID{0u}; chunk_id < _in_table->chunk_count(); ++chunk_id) {
-      jobs.push_back(_create_job_and_schedule(chunk_id, output_mutex));
+      if (_in_table->get_chunk(chunk_id)) {
+        jobs.push_back(_create_job_and_schedule(chunk_id, output_mutex));
```

Comment:
This still looks unsafe. Just because the pointer exists right now, it is not guaranteed to exist later.

Reply:
Correct. There should be an additional check in the job itself.